src/cgutils.cpp

```diff
@@ -1075,6 +1075,9 @@ static Value *emit_typeptr_addr(Value *p)
     return emit_nthptr_addr(p, -offset);
 }
 
+static Value *boxed(const jl_cgval_t &v, jl_codectx_t *ctx, bool gcooted=true);
+static Value *boxed(const jl_cgval_t &v, jl_codectx_t *ctx, jl_value_t* type) = delete; // C++11 (temporary to prevent rebase error)
```

Comment:
> We only need a pre-C++11 compiler when building against LLVM 3.3

We don't "need" pre-C++11 compiler, we just don't require a c++11 mode. All the compiler we support should support c++11 and we should just bump the compiler flags in `Make.inc` to enable it.